src/commands/debug.js

```diff
@@ -0,0 +1,247 @@
+      print.error(`No block number provided`)
+      print.info(HELP)
+      process.exitCode = 1
+      return
+    }
+
+    // Validate node
+    if (!node) {
+      print.error(`No Graph node provided`)
+      print.info(HELP)
+      process.exitCode = 1
+      return
+    }
+
+    // Validate IPFS
+    if (!ipfs) {
+      print.error(`No IPFS node provided`)
+      print.info(HELP)
+      process.exitCode = 1
+      return
+    }
+
+    let protocol
+    try {
+      // Checks to make sure deploy doesn't run against
+      // older subgraphs (both apiVersion and graph-ts version).
+      //
+      // We don't want the deploy to run without these conditions
+      // because that would mean the CLI would try to compile code
+      // using the wrong AssemblyScript compiler.
+      await assertManifestApiVersion(manifest, '0.0.5')
+      await assertGraphTsVersion(path.dirname(manifest), '0.22.0')
+
+      const dataSourcesAndTemplates = await DataSourcesExtractor.fromFilePath(manifest)
+
+      protocol = Protocol.fromDataSources(dataSourcesAndTemplates)
+    } catch (e) {
+      print.error(e.message)
+      process.exitCode = 1
+      return
+    }
+
+    const isLocal = node.match(/localhost/) || node.match(/127\.0\.0\.1/)
+    if (!isLocal) {
+      print.error(`The Graph node must be running locally`)
+      process.exitCode = 1
+      return
+    }
+
+    let compiler = createCompiler(manifest, {
+      ipfs,
+      outputDir,
+      outputFormat: 'wasm',
+      skipMigrations: true,
+      blockIpfsMethods: false,  // Network does not support publishing subgraphs with IPFS methods
+      protocol,
+    })
+
+    // Exit with an error code if the compiler couldn't be created
+    if (!compiler) {
+      process.exitCode = 1
+      return
+    }
+
+    let requestUrl = new URL(node)
+    let client = createJsonRpcClient(requestUrl)
+
+    // Exit with an error code if the client couldn't be created
+    if (!client) {
+      process.exitCode = 1
+      return
+    }
+
+    let debugSubgraph = async ipfsHash => {
+      let spinner = print.spin(`Debugging on Graph node ${requestUrl}`)
+
+      client.request(
+        'subgraph_debug',
+        {name: subgraphName, ipfs_hash: ipfsHash, endpoint: endpoint, block: block},
+        async (requestError, jsonRpcError, res) => {
+          if (jsonRpcError) {
+            spinner.fail(
+              `Failed to debug on Graph node ${requestUrl}: ${jsonRpcError.message}`,
+            )
+
+            // Provide helpful advice when the subgraph has not been created yet
+            if (jsonRpcError.message.match(/subgraph name not found/)) {
+              print.info(`
+Make sure to create the subgraph first by running the following command:
+$ graph create --node ${node} ${subgraphName}`)
+            }
+            process.exitCode = 1
+          } else if (requestError) {
+            spinner.fail(`HTTP error deploying the subgraph ${requestError.code}`)
+            process.exitCode = 1
```

Comment:
Are there any other messages besides the regular `deploy` ones that we would like to map here?

Reply:
No, not really.